[doc] Add the spelling checker in the install guide (#7193)
PyCQA/pylint: doc/user_guide/installation/command_line_installation.rst

   pip install pylint

Or if you want to also check spelling with ``enchant`` (you might need to
`install the enchant C library <https://pyenchant.github.io/pyenchant/install.html#installing-the-enchant-c-library>`_):

.. code-block:: sh

   pip install pylint[spelling]

> The newest pylint support python interpreters that are not past end of life.

Reviewer: ```suggestion
The newest pylint supports all Python interpreters that are not past end of life.
```

We recommend to use the latest interpreter because we rely on the ``ast`` builtin

> module that get better with each new python interpreter. For example a python

Reviewer: ```suggestion
module that gets better with each new Python interpreter. For example a Python
```

> 3.6 interpreter can't analyse 3.8 syntax (because of the new walrus operator) while a 3.8

Reviewer: ```suggestion
3.6 interpreter can't analyse 3.8 syntax (amongst others, because of the new walrus operator) while a 3.8
```

> interpreter can deal with python 3.6. See :ref:`using pylint with multiple interpreters <continuous-integration>` for more detail.

Reviewer: ```suggestion
interpreter can also deal with Python 3.6. See :ref:`using pylint with multiple interpreters <continuous-integration>` for more details.
```

   pip install pylint

Or if you want to also check spelling with ``enchant`` (you might need to

> `install the enchant C library <https://pyenchant.github.io/pyenchant/install.html#installing-the-enchant-c-library>`_ ):

Reviewer: ```suggestion
`install the enchant C library <https://pyenchant.github.io/pyenchant/install.html#installing-the-enchant-c-library>`_):
```

.. code-block:: sh

   pip install pylint[spelling]

> Use the newest Python interpreter if you can.
> 

Reviewer: ```suggestion
```

This is such a weird tip. I think that is really something for the user to decide 😄

Author: It's in the context of supporting 3.5 to 3.9, you should use pylint with python 3.9 and set ``py-version`` to 3.5.

Reviewer: Yeah, but shouldn't that be in the configuration section somewhere? This feels a bit out of place.

Author: We're talking about what interpreter to use during installation. Maybe we should group this sentence it in the paragraph just under this so it's clearer ? I don't know if most user maintain multiple interpreters though, so we could move this on the assumption that most user just use their current interpreter ?

Reviewer: > I don't know if most user maintain multiple interpreters though, so we could move this on the assumption that most user just use their current interpreter ?

Yeah I think this is a fair assumption. If users use different interpreters they likely also know that it is better to use the most recent interpreter for their tools.
